weatherreport.cpp

```diff
@@ -36,6 +36,24 @@ class SensorStub : public IWeatherSensor {
     }
 };
 
+class SensorStub2 : public IWeatherSensor {
```

Comment:
Creating a new class indeed works to exercise different behaviors in the code. You could also make the class to accept the values it needs to return (in its constructor). Then we can make more variants with objects, instead of making new classes for every variant